drivers/SmartThings/wemo/src/init.lua

```diff
@@ -121,32 +89,39 @@ local function device_init(driver, device)
     discovery.find(device.device_network_id, function(found) info = found end)
     if info then break end
     local tm = backoff()
-    log.debug("Failed to initialize device, retrying after delay", tm)
+    device.log.info_with({ hub_logs = true }, string.format("Failed to initialize device, retrying after delay: %.1f", tm))
     socket.sleep(tm)
   end
 
   if not info or not info.ip or not info.serial_num then
-    log.warn_with({ hub_logs = true }, "[" .. device.device_network_id .. "] device not found on network")
-    device:offline() -- Mark device as being unavailable/offline
+    device.log.error_with({ hub_logs = true }, "device not found on network")
+    device:offline()
     return
   end
 
   device:online()
 
-  --ip and port are persisted since sometimes wemo just stop responding to ssdp even though
-  --they are connected to the network. In this case we want them to continue to function
-  --across driver restarts.
+  --Sometimes wemos just stop responding to ssdp even though they are connected to the network.
+  --Persist to avoid issues with driver restart
   device:set_field("ip", info.ip, { persist = true })
   device:set_field("port", info.port, { persist = true })
   device:set_field("serial_num", info.serial_num, { persist = true })
 
-  --TODO maybe we should call_on_schedule with the device thread, and the polling.
-  -- instead of doing the resubscribe for all devices at once.
-  protocol.subscribe(server, device)
-end
+  local jitter = 2 * math.random()
```

Comment:
What's the variance for?

Reply:
During Migration, driver startup, and polling intervals, it spreads network traffic out. If 10 wemo devices are added, we would send 10 http request in like a quarter second if all devices respond to the first discovery message. And then every 60 sec we would send 10 more. This is an attempt to spread that out a little bit. The same goes for re subscribing. It makes the biggest difference during driver restart.

Im not sure it solves a problem that exists, but these devices are unreliable and I want to give them the best chance.